src/material-table.js

```diff
@@ -82,7 +82,35 @@ export default class MaterialTable extends React.Component {
           : '';
     }
 
-    this.dataManager.setColumns(props.columns, prevColumns);
+    const columnsCopy = JSON.parse(JSON.stringify(props.columns));
```

Comment:
Can we do the copy inside the if to reduce work? And wrap it with an else if not?

Reply:
This solves an unreported bug with shared column defintions and groupings :)

https://user-images.githubusercontent.com/19377375/125429648-f4851150-698e-4066-ae95-47af38d13d55.mov


```tsx
const columns = [{
    title: 'a',
    field: 'a',
},{
    title: 'b',
    field: 'b',
}];
const data = [{a: 'a', b:'b'}]
return (
<MaterialTable
          title="Basic"
          columns={columns}
          data={data}
          options={{
              grouping: true,
          }}
      />
      <MaterialTable
          title="Basic"
          columns={columns}
          data={data}
          options={{
              grouping: true,
          }}
      />)
```